contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -281,26 +304,80 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         emit ProgramEnabled({ pool: pool, status: status, remainingRewards: p.remainingRewards });
     }
 
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function setAutoProcessRewardsCount(uint256 newAutoProcessRewardsCount)
+        external
+        greaterThanZero(newAutoProcessRewardsCount)
+        onlyAdmin
+    {
+        uint256 prevAutoProcessRewardsCount = _autoProcessRewardsCount;
+        if (prevAutoProcessRewardsCount == newAutoProcessRewardsCount) {
+            return;
+        }
+
+        _autoProcessRewardsCount = newAutoProcessRewardsCount;
+
+        emit AutoProcessRewardsCountUpdated({
+            prevAutoProcessRewardsCount: prevAutoProcessRewardsCount,
+            newAutoProcessRewardsCount: newAutoProcessRewardsCount
+        });
+    }
+
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function autoProcessRewards() external nonReentrant {
+        uint256 numOfPools = _pools.length();
+        uint256 index = _nextProcessRewardsIndex;
+        uint256 count = _autoProcessRewardsCount;
+        uint256 maxCount = Math.min(count * 2, numOfPools);
+
+        for (uint256 i = 0; i < maxCount; i++) {
+            index = (index + i) % numOfPools;
+            if (_processRewards(Token(_pools.at(index)), true)) {
+                count -= 1;
+                if (count == 0) {
+                    break;
+                }
+            }
+        }
+
+        _nextProcessRewardsIndex = index + 1;
+    }
+
     /**
      * @inheritdoc IAutoCompoundingRewards
      */
     function processRewards(Token pool) external nonReentrant {
+        _processRewards(pool, false);
+    }
+
+    /**
+     * @dev processes the rewards of a given pool
```

Comment:
"and returns whether the rewards processing was successfully triggered or skipped" (or something like that)

Reply:
done